music_metadata_extractor/spotify.py

```diff
@@ -115,8 +115,10 @@ def search(**kwargs) -> ProviderData:
         )
     else:
         response = SPOTIFY_CLIENT.search(q=f"{kwargs['title']}", type="track", limit=1)
-    track_data = response["tracks"]["items"][0]
-    if not track_data:
+    
+    try:
+        track_data = response["tracks"]["items"][0]
+        response = parse_track_response(track_data)
+        return response
+    except:
         raise IndexError("No data found in Spotify")
```

Comment:
I suppose you might have to [raise the exception properly](https://stackoverflow.com/a/6246394/4396392) to catch and log the stack trace. Can you please make the relevant changes in all methods where relevant?

Reply:
Sure. This article seems interesting.